src/main/java/net/sf/jabref/logic/importer/fetcher/CrossRef.java

```diff
@@ -84,33 +90,37 @@ private static String enhanceQuery(String query, BibEntry entry) {
         return enhancedQuery.toString();
     }
 
-    private static boolean checkValidity(BibEntry entry, JSONArray result) {
+    private static Optional<String> findMatchingEntry(BibEntry entry, JSONArray results) {
         final String entryTitle = REMOVE_BRACES_FORMATTER.format(entry.getLatexFreeField(FieldName.TITLE).orElse(""));
 
-        // currently only title-based
-        // title: [ "How the Mind Hurts and Heals the Body." ]
-        // subtitle: [ "" ]
-        try {
-            // title
-            JSONObject data = result.getJSONObject(0);
-            String dataTitle = data.getJSONArray("title").getString(0);
-
-            if (editDistanceIgnoreCase(entryTitle, dataTitle) <= METRIC_THRESHOLD) {
-                return true;
-            }
-
-            // subtitle
-            // additional check, as sometimes subtitle is needed but sometimes only duplicates the title
-            if (data.getJSONArray("subtitle").length() > 0) {
-                String dataWithSubTitle = dataTitle + " " + data.getJSONArray("subtitle").getString(0);
-
-                return editDistanceIgnoreCase(entryTitle, dataWithSubTitle) <= METRIC_THRESHOLD;
+        for (int i = 0; i < results.length(); i++) {
+            // currently only title-based
+            // title: [ "How the Mind Hurts and Heals the Body." ]
+            // subtitle: [ "" ]
+            try {
+                // title
+                JSONObject data = results.getJSONObject(i);
+                String dataTitle = data.getJSONArray("title").getString(0);
+
+                if (editDistanceIgnoreCase(entryTitle, dataTitle) <= METRIC_THRESHOLD) {
+                    return Optional.of(data.getString("DOI"));
+                }
+
+                // subtitle
+                // additional check, as sometimes subtitle is needed but sometimes only duplicates the title
+                if (data.getJSONArray("subtitle").length() > 0) {
+                    String dataWithSubTitle = dataTitle + " " + data.getJSONArray("subtitle").getString(0);
+
+                    if (editDistanceIgnoreCase(entryTitle, dataWithSubTitle) <= METRIC_THRESHOLD) {
+                        return Optional.of(data.getString("DOI"));
+                    }
+                }
+            } catch(JSONException ex) {
+                return Optional.empty();
```

Comment:
This should be detected by our Tests. Shouldn't it?

However, logging here does no harm.